src/scenes/ProgressReports/Detail/ProgressReportDrawer/RoleIcon.tsx

```diff
@@ -0,0 +1,56 @@
+import { Grading, Translate } from '@mui/icons-material';
+import { Box } from '@mui/material';
+import { Role } from '~/api/schema.graphql';
+import { PeopleJoinedIcon } from '~/components/Icons';
+import { ProjectManagerIcon } from '~/components/Icons/ProjectManagerIcon';
+import { colorPalette } from './colorPalette';
+
+interface StepIconProps {
+  roleStep?: Role | null;
+}
+
+const variantToIconMapper: {
+  [key in Role]: React.FC | null;
```

Comment:
I'd be ok with this then we could omit all the roles that we don't want to declare icons for. And it doesn't need adjusting if a role is added.

We could also enforce that the map is `SvgIcons`, then we could skip the `Box` below.
```suggestion
  [key in Role]?: SvgIconComponent;
```

```ts
const Icon = variantToIconMapper[role];
return Icon && <Icon sx={...} />;
```